### PR TITLE
Update Dotty (Scala 3) to 3.0.0-M1

### PR DESCRIPTION
**Pull request by atrianac**

### CODE REVIEW

**Comment by sideeffffect** [comment]
@atrianac could you please fix the conflict? :pray:

**Comment by atrianac** [comment]
> @atrianac could you please fix the conflict? 🙏

Just performing the rebase

**Comment by adamgfraser** [comment]
@atrianac Looks like this just needs a format.

File: build.sbt [hunk 1]
```diff
@@ -37,7 +37,12 @@ addCommandAlias(
   ";coreNative/test:compile"
 )
 
-val zioVersion = "1.0.3"
+val zioVersion = "1.0.4"
+libraryDependencies ++= Seq(
```
Comment (sideeffffect) [inline build.sbt:41]: We shouldn't need to add any `libraryDependencies` here, merely updating `zioVersion` should be enough

**Review by sideeffffect — CHANGES_REQUESTED** [review]
Just one small thing. Thanks!
(And than we have to wait for ZIO release, of course :joy: )

File: build.sbt (hunk 1 above)
Comment (atrianac) [reply build.sbt]: > We shouldn't need to add any `libraryDependencies` here, merely updating `zioVersion` should be enough

Done! Thanks

**Comment by jdegoes** [comment]
@atrianac You still looking to get this merged in? Just needs to  merge conflicts and format the code, looks like. Let me know! I will close for now but please re-open as your time and interest allows.

**Comment by sideeffffect** [comment]
We'll waiting here on ZIO to make a release, then we can use the newest Dotty version.

**Pull request closed without merging**